### ISSUE TITLE
Add the ability to reduce the number of recent events in the Recent Actions section

### ISSUE THREAD

**Issue by Vstory** [AdguardTeam/AdguardForAndroid#5407]
### Issue Details

AdGuard's recent filter log viewing interface, the number of saved events is too large, which will cause the scrolling viewing page to freeze. The current value is 10000, it is recommended to change it to 1000.

Of course, it would be better to allow users to customize the size in the low-level settings.

### Proposed solution

The current value is 10000, it is recommended to change it to 1000.

Of course, it would be better to allow users to customize the size in the low-level settings.

### Alternative solution

![Screenshot_2024-07-25-20-15-42-430_com adguard android](https://github.com/user-attachments/assets/d52031b9-d0a9-4964-b4a3-41a6d65eb533)

**Comment by maxikuzmin** [comment]
@Vstory as far as I understand, 10000 was configured to be more convenient to use, as there may be too many requests in a short period of time, some requests may not fit into the number 1000. It will not be too comfortable if you see requests in a short period of time when using recent activity. Why would you want to reduce to 1000?

**Comment by Vstory** [comment]
> @Vstory as far as I understand, 10000 was configured to be more convenient to use, as there may be too many requests in a short period of time, some requests may not fit into the number 1000. It will not be too comfortable if you see requests in a short period of time when using recent activity. Why would you want to reduce to 1000?

The number 1000 is just my personal opinion. As you said, there are too many requests in a short period of time. There are too many requests on this page in a short period of time, which will cause the page to be very stuck when scrolling.My opinion is whether changing the value from 10000 to 1000 will reduce the page lag and allow for smoother viewing of the page content.

**Comment by maxikuzmin** [comment]
@Vstory I suggest having the choice of displaying the last 10000 requests or 1000. We can change your issue to suggest such a feature to see how many people would vote for it. What do you think?

**Comment by Vstory** [comment]
> @Vstory I suggest having the choice of displaying the last 10000 requests or 1000. We can change your issue to suggest such a feature to see how many people would vote for it. What do you think?

Yes, 10,000 recent requests are too large, so only 1,000 are needed for rule interception preview. After all, this is only the most recent filter. If you want to see the complete filter, you can grab the log.
By reducing the number of displays, I hope this will improve the smoothness of this page.

**Comment by maxikuzmin** [comment]
@Vstory okay, let's see how many likes your feature request gets
We monitor feature requests regularly to review and take on feature requests with the most likes

**Comment by vonDubenshire** [comment]
> @Vstory okay, let's see how many likes your feature request gets We monitor feature requests regularly to review and take on feature requests with the most likes

I had not even thought about this as I thought it was annoying but just something that existed after AdGuard was rebuilt and Assistant on Android relaunched. 

While I understand the mass number of requests maked 10,000 seem reasonable, I honestly have never had an issue going back that far and if I did I'd likely be logging & debugging, if not adding for help from you guys at that point.

When I look at my recent activity, there's a good chance it will be very annoyingly slow for at least a few seconds on most devices if I've been browsing heavily.  It doesn't cause problems but it has made me mis-click a few times by forgetting that there's going to be a bit of lag before I need to try to find what I'm looking for. 

Even 5000 or a deep level setting that would allow 2000 or 10000 would work for me.

**Comment by Vstory** [comment]
In daily use, I always turn off the log of this option to save power. I only enable log output when an exception occurs on the filtering page.
